content/intro-to-storybook/react/ja/screen.md

```diff
@@ -0,0 +1,181 @@
+
+しかし、`PureInboxScreen` には問題があります。`PureInboxScreen` が表示用コンポーネントであっても、その子供である `TaskList` は表示用ではないのです。つまり、`PureInboxScreen`が「コンテナー化」により汚染されたと言えます。なので、`InboxScreen.stories.js` を以下のようセットアップすると:
+
+```javascript
+// src/components/InboxScreen.stories.js
+
+import React from 'react';
+
+import { PureInboxScreen } from './InboxScreen';
+
+export default {
+  component: PureInboxScreen,
+  title: 'InboxScreen',
+};
+
+export const Default = () => <PureInboxScreen />;
+
+export const Error = () => <PureInboxScreen error="Something" />;
+```
+
+`Error` ストーリーは正しく動いていますが、`Default` ストーリーには問題があります。それは `TaskList` に接続するべき Redux のストアがないためです。(同様に `PureInboxScreen` を単体テストしようとしても同じことが起こります。)
+
+![壊れている Inbox](/intro-to-storybook/broken-inboxscreen.png)
+
+この問題を回避する方法の 1 つは、コンテナーコンポーネントをアプリケーションの最上位にのみ描画し、代わりにコンポーネント階層の下層に必要なデータをすべて上位のコンポーネントから渡すことです。
+
+ですが、開発では**きっと**コンポーネント階層の下位の層でコンテナーを描画する必要が出てくるでしょう。アプリケーション全体、もしくは大部分を Storyook で描画したいなら、解決策が必要です。
+
+<div class="aside">
+補足として、データを下の階層に渡していくことは正当な手法です。<a href="http://graphql.org/">GraphQL</a> を使う場合は特に。<a href="https://www.chromatic.com">Chromatic</a> を作る際にはこの手法で 800 以上のストーリーを作成しました。
+</div>
+
+## デコレーターを使用してコンテキストを渡す
+
+ストーリーの中で `InboxScreen` に Redux のストアを渡すのは簡単です！モック化したストアをデコレーター内部で使用します:
+
+```javascript
+// src/components/InboxScreen.stories.js
+
+import React from 'react';
+import { action } from '@storybook/addon-actions';
+import { Provider } from 'react-redux';
+
+import { PureInboxScreen } from './InboxScreen';
+import { defaultTasksData } from './TaskList.stories';
+
+export default {
+  component: PureInboxScreen,
+  title: 'InboxScreen',
+  decorators: [story => <Provider store={store}>{story()}</Provider>],
+};
+
+// とても簡素な Redux ストアのモック
+const store = {
+  getState: () => {
+    return {
+      tasks: defaultTasksData,
+    };
+  },
+  subscribe: () => 0,
+  dispatch: action('dispatch'),
+};
+
+export const Default = () => <PureInboxScreen />;
+
+export const Error = () => <PureInboxScreen error="Something" />;
+```
+
+同様に [Apollo](https://www.npmjs.com/package/apollo-storybook-decorator) や [Relay](https://github.com/orta/react-storybooks-relay-container) など、他のデータライブラリー向けのモックコンテキストも存在します。
+
+Storybook で状態を選択していくことで、問題なく出来ているか簡単にテストできます:
+
+<video autoPlay muted playsInline loop >
+
+  <source
+    src="/intro-to-storybook/finished-inboxscreen-states.mp4"
+    type="video/mp4"
+  />
+</video>
+
+## コンポーネント駆動開発
+
+まず、一番下の `Task` から始めて、`TaskList` を作り、画面全体の UI が出来ました。`InboxScreen` ではネストしたコンテナーコンポーネントを含み、一緒にストーリーも作成しました。
+
+<video autoPlay muted playsInline loop style="width:480px; height:auto; margin: 0 auto;">
+  <source
+    src="/intro-to-storybook/component-driven-development-optimized.mp4"
+    type="video/mp4"
+  />
+</video>
+
+[**コンポーネント駆動開発**](https://blog.hichroma.com/component-driven-development-ce1109d56c8e) (CDD) はコンポーネント階層を上がるごとに少しずつ複雑性を拡張していきます。利点としては、開発プロセスに集中できること、UI の組み合わせの網羅性を向上できること、が挙げられます。要するに、CDD によって、高品質で複雑な UI を作ることができます。
```

Comment:
@chameleonhead when you're able can you please adjust the link https://blog.hichroma.com/component-driven-development-ce1109d56c8e to https://www.componentdriven.org/